### PR TITLE
gossip: shred version discovery

### LINKED ISSUE #1777: Fix shred version discovery for gossip
Gossip currently picks up the shred version from the very first contact info it receives instead of checking for the right entrypoint.

### PR DESCRIPTION
**Pull request by arjain4**
Closes #1777

### CODE REVIEW

File: src/app/fdctl/config.c [hunk 1]
```diff
@@ -142,6 +142,7 @@ static int parse_key_value( config_t *   config,
       }                                                                              \
       config->esection edot ekey[ config->esection edot ekey##_cnt ] = (uint)result; \
       config->esection edot ekey##_cnt++;                                            \
+      return 1;                                                                        \
```
Comment (mmcgee-jump) [inline src/app/fdctl/config.c:145]: ???

**Review by mmcgee-jump — CHANGES_REQUESTED** [review]
.

File: src/app/fdctl/config.c [hunk 2]
```diff
@@ -349,6 +346,7 @@ config_parse_array( const char * path,
                     char * key,
                     int * in_array,
                     char * value ) {
+  if( *in_array ) value++;
```
[the remark this answers is not in the document]
Comment (arjain4) [reply src/app/fdctl/config.c:349]: these are bug fixes

File: src/app/fdctl/config.c (hunk 1 above)
Comment (arjain4) [reply src/app/fdctl/config.c:145]: bug fix, it needs to return 1 like all other cases otherwise it errors

Comment (lheeger-jump) [inline src/app/fdctl/config.c:145]:
```suggestion
      config->esection edot ekey##_cnt++;                                            \
      return 1;                                                                      \
```

**Review by lheeger-jump — APPROVED** [review]

**Review by asiegel-jt — APPROVED** [review]

**Pull request merged**